be/src/exprs/agg/approx_top_k.h

```diff
@@ -0,0 +1,466 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <type_traits>
+
+#include "column/array_column.h"
+#include "column/column_hash.h"
+#include "column/column_helper.h"
+#include "column/struct_column.h"
+#include "column/type_traits.h"
+#include "exprs/agg/aggregate.h"
+#include "exprs/agg/aggregate_traits.h"
+#include "runtime/mem_pool.h"
+#include "types/logical_type.h"
+#include "util/phmap/phmap.h"
+
+namespace starrocks {
+
+template <LogicalType LT>
+struct ApproxTopKState {
+    using CppType = RunTimeCppType<LT>;
+    using ColumnType = RunTimeColumnType<LT>;
```

Comment:
should we imp a general version for comple types? like array, map, struct?

Reply:
This is beyond the scope of this pr. May be we will support semi-structured type.